### PR TITLE
New FMDB spec, config updates in support of it

### PR DESCRIPTION
**Pull request by khawkins**
Adding our own spec for FMDB, so we can support our own build of SQLCipher.

Note: Ordering of `source` tags in your Podfile matters.  With this setup, if you include our specs repo first, i.e.:

```
source 'https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Specs.git'
source 'https://github.com/CocoaPods/Specs.git'
```

You will get our version of FMDB and SQLCipher.  If you flip the ordering, you'll get the stock version of FMDB and SQLCipher published to cocoapods.org.

### CODE REVIEW

File: FMDB/4.0.1/FMDB.podspec [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                     :submodules => true }
+
+  s.requires_arc = true
+  s.default_subspec  = 'SQLCipher'
+
+  s.subspec 'SQLCipher' do |sqlcipher|
+
+      sqlcipher.source_files = 'external/fmdb/src/fmdb/FM*.{h,m}'
+      sqlcipher.exclude_files = 'external/fmdb/src/fmdb.m'
+      sqlcipher.xcconfig = { 'OTHER_CFLAGS' => '$(inherited) -DSQLITE_HAS_CODEC -DHAVE_USLEEP=1' }
```
Comment (bhariharan) [inline FMDB/4.0.1/FMDB.podspec]: What's this new sleep flag?

**Comment by bhariharan** [comment]
:+1: Merge away!

File: FMDB/4.0.1/FMDB.podspec (hunk 1 above)
Comment (khawkins) [inline FMDB/4.0.1/FMDB.podspec]: I dunno.  It's in the main FMDB spec, so I just kept it.

Comment (khawkins) [inline FMDB/4.0.1/FMDB.podspec]: Looking through the FMDB code, it's not used at all, and is a flag that's meant for builds of SQLite.  So there's really no good reason it's there.  I'm going to take it out.

**Pull request merged**